### ISSUE TITLE
Clarify enabled effects and presets

### ISSUE THREAD

**Issue by korpe1** [wwmm/easyeffects#3142]
Could we make the UI more clear about which effects and presets are enabled? I suggest to use checkbox or slide switch elements to enable things.

 * The arrow icon is always highlighted and its bigger element than the text.
 * Highlighted text is a pretty popular way to indicate the location in menus; i.e. in the picture user could think he is in the Equalizer view.

![image](https://github.com/wwmm/easyeffects/assets/145599023/f94e3da4-580f-4094-9806-08a3416f46aa)

And same problems in the preset selector where a small text is the only indicator:
![image](https://github.com/wwmm/easyeffects/assets/145599023/c47c04b2-6cea-45f9-adc8-53bf609a4986)

**Comment by wwmm** [comment]
>Could we make the UI more clear about which effects and presets are enabled?

Just to avoid any misunderstanding EasyEffects does not operate around the idea of "enabled presets". Imagine that you've loaded a preset and manually changed a parameter after this. EasyEffects database will remember the current value and use it from this point on. It does not matter that a preset was loaded before.  Presets are just a convenient way to load a given set of values. Our database does not work based on the preset file. When I added the preset name in the menu the idea was more along the lines of "last used preset". Nothing more than that.

>And same problems in the preset selector where a small text is the only indicator:

I never wanted to highlight the preset row for the same reasons I talked about above. EasyEffects does not keep track of loaded presets because they are just a fast way to load values. Our database does not use them. The preset selected there may not represent the current values depending on what the user is doing. 

>The arrow icon is always highlighted and its bigger element than the text.

The original intention was having this icon representing the flow of audio buffers in the pipeline. Disabling the plugin through that toggle button puts it in passthrough mode. So audio is still flowing through it. It just do not modify the data. it is not a big deal to me to try to put the icon in passive mode as long as people do not also think that the plugin is totally removed from the pipeline.

**Comment by Digitalone1** [comment]
Please, if you can, try out the master branch, the preset menu is changed and now we show the "the last loaded preset".

As @wwmm said, effects are not enabled/disabled, but they are dimmed when bypassed.

**Comment by korpe1** [comment]
Okay, the reasons for presets make sense.

> The original intention was having this icon representing the flow of audio buffers in the pipeline. Disabling the plugin through that toggle button puts it in passthrough mode. So audio is still flowing through it. It just do not modify the data. It is not a big deal to me to try to put the icon in passive mode as long as people do not also think that the plugin is totally removed from the pipeline.

Okay, that is important background, but I think we should allow user keep set of effects on the side panel regardless are them disabled or not. And even without that, the status need to be more clear. I keep EQ always open, but I wan't to know is it on or not. Now the indicator is too unnoticeable. Could we make a dedicated indicator for the statuses of effects?

**Comment by Digitalone1** [comment]
> Now the indicator is too unnoticeable.

If you look the indicator, you are looking at the wrong thing. You should look at the effect name. I can see in the screenshot that only the first Equalizer is not bypasses. Then, yes, maybe we can make also the arrow as dimmed in bypass mode, but I don't think it's a good idea.

When we added the player/device rows (they were missing initially) we wanted to give an hint of the "flow", but I made the icon dimmed because that row is not interactive. If you dim also the arrow, it seems not interactive too, which it's not the case since it's an effect.

**Comment by violetmage** [comment]
For displaying the bypass/active state of the effects, I think that instead of dimming bits of the UI, we should simply have a different icon that represents the bypass state.

Perhaps an arrow with a dashed tail, instead of solid.

This way the communication of "flow" still happens, but it includes the bypass state as well. To me, this seems much better than relying on the relatively low-contrast dimming of text.

**Comment by wwmm** [comment]
> For displaying the bypass/active state of the effects, I think that instead of dimming bits of the UI, we should simply have a different icon in place of the flow arrow that represents the bypass state.
> 
> Perhaps an arrow with a dashed tail, instead of solid.
> 
> This way the communication of "flow" still happens, but it includes the bypass state as well. To me, this seems much better than relying on the relatively low-contrast dimming of text.

Assuming we find a good icon it will probably be the most reasonable thing to do. But I would still dim the label contrast.

**Comment by korpe1** [comment]
> For displaying the bypass/active state of the effects, I think that instead of dimming bits of the UI, we should simply have a different icon in place of the flow arrow that represents the bypass state.

Exactly! I just want a more noticeable indicator; was it contrast, text or an icon doesn't matter.

> Perhaps simply an arrow with a dashed tail, instead of solid.

I think that's also a good idea.

**Comment by Digitalone1** [comment]
@korpe1 I don't think there's an icon like that. Please suggest one that has the symbolic style (changes the color in light/dark theme).

**Comment by Digitalone1** [comment]
Maybe it's just better to change the style. The red `error` could be too much. Maybe the yellow `warning`...

**Comment by Digitalone1** [comment]
![image](https://github.com/wwmm/easyeffects/assets/25790525/acd31810-3a4c-4ba5-aaa3-682d15fff3f5)

What do you think? 

I picked these icons: 
- https://www.svgrepo.com/svg/85713/arrow-down
- https://www.svgrepo.com/svg/175958/parallel-lines

[CC0 license](https://creativecommons.org/public-domain/cc0/), we can use them.

**Comment by LebedevRI** [comment]
@wwmm @Digitalone1 have https://www.svgrepo.com/svg/503241/bypass been considered for bypass icon?
The "pause" looks a bit out of place IMHO...

**Comment by Digitalone1** [comment]
> @wwmm @Digitalone1 have https://www.svgrepo.com/svg/503241/bypass been considered for bypass icon?

No, but it could be used.

**Comment by Digitalone1** [comment]
I tried to test it. Also made a new version for the plugin bypass, rotating the original one, since the flow in the plugin list is from top to bottom.

This is the result:

![image](https://github.com/wwmm/easyeffects/assets/25790525/ba2c5f77-46fa-44d2-861f-16e6442ab875)

Which I don't like. The icons overall and their arrows are small and that does not help to express their meaning. 

The previous one (which are two parallel lines, not the pause icon) may be out of place, but it's still better than this new one, IMHO.

It should be also noted that the icon does not express the order, otherwise we could use a numbered list. The icon means that the audio stream is flowing through the filter/plugin from top to bottom. I intended the two parallel lines as "the audio is going down on the track without passing into the filter". If you intend it as a pause, it's still pertinent to its meaning since the plugin is in passthrough mode (it's not operating).

**Comment by wwmm** [comment]
>The icons overall and their arrows are small and that does not help to express their meaning.

And for some reason it does not align horizontally to form a straight line with the icons in the other rows. Even if the size issue is somehow solved a vertical variant would probably have to be done through changes to the svg icon. Just rotating is clearly not going to work.

At least for now I think it is better to keep the current icon.

**Comment by wwmm** [comment]
And at least for me another side effect of the small curved arrow size is that the horizontal icon in the global bypass button makes me thing about an eye instead of a bypass :smile: .

**Comment by korpe1** [comment]
I will give a more precise feedback later, but now I just say: great ideas! Thank you!
